src/main/scala/com/lightbend/sbt/MultiReleaseJarPlugin.scala

```diff
@@ -35,19 +35,69 @@ object MultiReleaseJarPlugin extends AutoPlugin {
   val MULTI_RELEASE_KEY = "Multi-Release"
   // ---------------------------------------
 
-  override def projectSettings = Seq(
-    
-    compile := 
+  def jdkVersion: String = System.getProperty("java.version")
+  
+  override def projectSettings: Seq[Def.Setting[_]] =
+    jdkVersion match {
+      case "9" =>
+        println(
+          scala.Console.BOLD + 
+            "[multi-release-jar plugin] Running using JDK9! " +
+            "Will include classes and tests that require JDK9." +
+            scala.Console.RESET)
+        jdk9ProjectSettings
+      case "1.8" | "8" => 
+        println(
+          scala.Console.BOLD + 
+            "[multi-release-jar plugin] Running using JDK 8, " +
+            "note that JDK9 classes and tests will not be included in compile/test runs." + 
+            scala.Console.RESET)
+        
+        Seq.empty
+      case _ => 
+        throw new IllegalStateException("Only JDK 8 or 9 is supported by this build, because of the mult-release-jar plugin.")
+    }
+  
+  def jdk9ProjectSettings: Seq[Def.Setting[_]] = Seq(
+    compile in Compile := 
       (compile in MultiReleaseJar).dependsOn(
         compile in Compile
       ).value,
     
+//    test := 
+//      (test in MultiReleaseJarTest).dependsOn(
+//        test in Test
+//      ).value,
+    
```

Comment:
Does:

```scala
test in Test := {
  val _ = (test in MultiReleaseJarTest).value
  (test in Test).value
}
```

work?

Reply:
Oh it does, thanks! Will change the other place too where I used this weird pattern